examples/Training/markup.py

```diff
@@ -332,44 +336,66 @@ def __init__(self, handler):
         self.addFilter(r'(http://[\.a-zA-Z_/]+)', 'url')
 
 
-PYHEADER = """#!/usr/bin/env python
-# -*- coding: utf-8 -*-
+class JavaParser(Parser):
+    """
+    A specific Parser that adds rules for Java
+    """
+    def __init__(self, handler):
+        Parser.__init__(self, handler)
+        self.addRule(SphinxSubtitleRule('//'))
+        self.addRule(SphinxCommentRule('*'))
+        self.addRule(CodeRule())
 
-#
-# Copyright (C) 2014 University of Dundee & Open Microscopy Environment.
-#                    All Rights Reserved.
-# Use is subject to license terms supplied in LICENSE.txt
-#
+        # self.addFilter(r'\*(.+?)\*', 'emphasis')
+        self.addFilter(r'(http://[\.a-zA-Z_/]+)', 'url')
 
-\"\"\"
-FOR TRAINING PURPOSES ONLY!
-\"\"\"""".split("\n")
+# Marker to find in the file to parse
+START_MARKER = "start-code"
+END_MARKER = "end-code"
 
 
 def check_header(file_lines, quiet=False):
     """
     Checks the first N lines of the file that they match
-    PYHEADER. Returns the number of lines which should be skipped.
+    START_MARKER. Returns the number of lines which should be skipped.
     """
     lines = []
     for line in file_lines:
-        idx = len(lines)
         lines.append(line)
-        try:
-            test = PYHEADER[idx]
-            if test.strip() != line.strip():
-                raise Exception("bad header. expected: '%s'. found: '%s'."
-                                % (line.strip(), test.strip()))
-        except IndexError:
-            if not quiet:
-                print "ok"
-            break
+        if START_MARKER in line:
+            return len(lines)
     return len(lines)
 
 
-if __name__ == "__main__":
+def parseMatlab(Parser):
+    """
+    Generates a doc page for the Matlab files.
+    """
+    files = [
```

Comment:
Might it be possible to avoid these associative arrays, for each language, by using a structured comment in the example source files for the titles and then just parse all the files in a designated directory for each language?